modules/autoscale/src/main.py

```diff
@@ -0,0 +1,264 @@
+import googleapiclient.discovery
+from google.cloud import secretmanager
+import functions_framework
+from cloudevents.http import CloudEvent
+
+from json import loads
+from logging import getLogger, basicConfig, INFO, DEBUG
+from xml.etree import ElementTree as et
+from xml.etree.ElementTree import Element
+from os import getenv
+import base64
+
+from panos.panorama import Panorama
+
+class ConfigureLogger:
+    def __init__(self, *args, **kwargs):
+        self.logger = getLogger(self.__class__.__name__)
+        basicConfig(format='%(asctime)s %(message)s')
+        self.logger.setLevel(INFO if getenv("logger_level") else DEBUG)
+
+
+class VMSeriesAutoscaling(ConfigureLogger):
+    def __init__(self, cloud_event: CloudEvent):
+        super().__init__()
+
+        self.panorama_address = getenv('PANORAMA_ADDRESS')
+        self.panorama2_address = getenv('PANORAMA2_ADDRESS')
+        self.project_id = getenv('PROJECT_ID')
+        
+        self.init_panorama_credentials()
+
+        self.main(cloud_event)
+
+    def init_panorama_credentials(self):
+        """
+        Helper function used to get secret from Secret Manager.
+
+        :return: None
+        """
+        secret_name = getenv('SECRET_NAME')
+
+        client = secretmanager.SecretManagerServiceClient()
+        name = f"projects/{self.project_id}/secrets/{secret_name}/versions/latest"
+        response = client.access_secret_version(name=name)
+        secret_str = response.payload.data.decode('UTF-8')
+        
+        self.panorama_username = secret_str.split("|", 1)[0]
+        self.panorama_password = secret_str.split("|", 1)[1]
```

Comment:
Variables used within a class should be instantiate in init function, you can achieve it like this ->
In line **30** add code `self.panorama_username, self.panorama_password = self.init_panorama_credentials()`
and at line **47** add code `return secret_str.split("|", 1)[0], secret_str.split("|", 1)[1]`

Reply:
Fixed